test/RFClassifyTest.ecl

```diff
@@ -0,0 +1,127 @@
+IMPORT $.^ AS LT;
+IMPORT ML_Core.Types AS CTypes;
+IMPORT LT.LT_Types AS Types;
+
+NumericField := CTypes.NumericField;
+DiscreteField := CTypes.DiscreteField;
+GenField := Types.GenField;
+errorProb := 0;
+wiCount := 1;
+numTrainingRecs := 1000;
+numTestRecs := 100;
+numTrees := 1;
+numVarsPerTree := 3;
+
+// Return TRUE with probability p
+prob(REAL p) := FUNCTION
+  rnd := RANDOM() % 1000000 + 1;
+  isTrue := IF(rnd / 1000000 <= p, TRUE, FALSE);
+  RETURN isTrue;
+END;
+
+// Test Pattern -- Ordinal variable X1 determines OP(X2, X3) => Y
+//                 OP (OR, AND, XOR, NOR), is determined as follows:
+//                 X1 < -50 => OR(X2, X3); -50 <= X1 < 0 => AND(X2, X3);
+//                 0 < X1 <= 50 => XOR(X2, X3); X1 >= 50 => NOR(X2, X3);
+
+dsRec := {UNSIGNED id, REAL X1, UNSIGNED X2, UNSIGNED X3, UNSIGNED Y};
+dsRec0 := {UNSIGNED id, UNSIGNED X1, UNSIGNED X2, UNSIGNED X3, UNSIGNED Y};
+dummy := DATASET([{0, 0, 0, 0, 0}], dsRec);
+dsRec make_data0(dsRec d, UNSIGNED c) := TRANSFORM
+  SELF.id := c;
+  // Pick random X1:  -100 < X1 < 100
+  r1 := __COMMON__(RANDOM());
+  r2 := __COMMON__(RANDOM());
+  r3 := __COMMON__(RANDOM());
+  SELF.X1 := r1%4;
+  // Pick random X2 and X3: Choose val between 0 and 1 and round to 0 or 1.
+  SELF.X2 := ROUND(r2%1000000 / 1000000);
+  BOOLEAN x2B := SELF.X2=1;
+  SELF.X3 := ROUND(r3%1000000 / 1000000);
+  BOOLEAN x3B := SELF.X3=1;
+  BOOLEAN y := MAP(SELF.X1 = 0 => x2B OR x3B, // OR
+                         SELF.X1 = 1 => x2B AND x3B, // AND
+                         SELF.X1 = 2 => (x2B OR x3B) AND (NOT (x2B AND x3B)), // XOR
+                         (NOT (x2B OR x3B)));  // NOR
+  SELF.Y := IF(y, 1, 0);
+END;
+dsRec make_data(dsRec d, UNSIGNED c) := TRANSFORM
+  SELF.id := c;
+  // Pick random X1:  -100 < X1 < 100
+  r1 := __COMMON__(RANDOM());
+  r2 := __COMMON__(RANDOM());
+  r3 := __COMMON__(RANDOM());
+  SELF.X1 := ROUND(r1%1000000 / 10000 * 2 - 100);
+  // Pick random X2 and X3: Choose val between 0 and 1 and round to 0 or 1.
+  SELF.X2 := ROUND(r2%1000000 / 1000000);
+  BOOLEAN x2B := SELF.X2=1;
+  SELF.X3 := ROUND(r3%1000000 / 1000000);
+  BOOLEAN x3B := SELF.X3=1;
+  BOOLEAN y := MAP(SELF.X1 < -50 => x2B OR x3B, // OR
+                         SELF.X1 >= -50 AND SELF.X1 < 0 => x2B AND x3B, // AND
+                         SELF.X1 >= 0 AND SELF.X1 < 50 => (x2B OR x3B) AND (NOT (x2B AND x3B)), // XOR
+                         (NOT (x2B OR x3B)));  // NOR
+  SELF.Y := IF(y, 1, 0);
+END;
+ds := NORMALIZE(dummy, numTrainingRecs, make_data(LEFT, COUNTER));
```

Comment:
I would encourage you to use: ds := DATASET(numTrainingRecs, make_data(COUNTER)); since you are not using the dummy dataset.  You can also distribute this work if desired.